packages/ui/feature-builder-form-controls/src/lib/components/new-piece-properties-form/piece-properties-form.component.ts

```diff
@@ -0,0 +1,153 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './piece-properties-form.component.html',
+})
+export class PiecePropertiesFormComponent
+  implements OnInit, OnChanges
+{
+  @Input({ required: true }) pieceMetaData: PieceMetadataModel;
+  @Input({ required: true }) actionOrTriggerName: string;
+  @Input({ required: true }) stepName: string;
+  @Input({ required: true }) flow: Pick<PopulatedFlow, 'id' | 'version'>;
+  @Input({ required: true }) webhookPrefix: string;
+  @Input({ required: true }) formPieceTriggerPrefix: string;
+  @Input({ required: true }) propertiesMap: PiecePropertyMap;
+  @Input({ required: true }) input: Record<string, any> = {};
+  @Input({ required: true }) customizedInputs: Record<
+    string,
+    boolean
+  > = {};
+  @Input({ required: true })
+  allConnectionsForPiece: DropdownOption<`{{connections['${string}']}}`>[];
+  @Input({ required: true }) form: UntypedFormGroup;
+  @Input({ required:true }) hideCustomizedInputs = false
+  @Output() customizedInputsChanged = new EventEmitter<{
+    propertyName:string;
+    value: boolean
+  }>();
+  @Output() formValueChange = new EventEmitter<{
+    input: Record<string, any>;
+    customizedInputs: Record<string, boolean | Record<string, boolean>>;
+    valid: boolean;
+  }>();
+  readonly PropertyType = PropertyType;
+  sortedPropertiesByRequired: PiecePropertyMap;
+  emitNewChanges$?: Observable<unknown>;
+  stepChanged$ = new BehaviorSubject('');
+  constructor(private fb:UntypedFormBuilder){}
+  ngOnInit(): void {
+    this.emitNewChanges$ = this.createChangesListener();
+  }
+  ngOnChanges(changes: SimpleChanges): void {
+    const properties = changes['propertiesMap'];
+    const stepName = changes['stepName'];
+    if (
+      properties?.firstChange ||
+      stepName?.firstChange ||
+      !deepEqual(properties?.currentValue, properties?.previousValue) ||
+      stepName?.currentValue !== stepName?.previousValue
+    ) {
+      this.initializeForm();
+      if(stepName?.currentValue !== stepName?.previousValue)
+      {
+        this.stepChanged$.next(stepName.currentValue);
+      }
+    }
+  }
+
+  private initializeForm() {
+    this.sortPropertiesByRequired();
+    this.buildForm();
+  }
+
+  private sortPropertiesByRequired() {
+    const requiredProperties: PiecePropertyMap = {};
+    const optionalProperties: PiecePropertyMap = {};
+    Object.entries(this.propertiesMap).forEach(([key, value]) => {
+      if (value.required) {
+        requiredProperties[key] = value;
+      } else {
+        optionalProperties[key] = value;
+      }
+    });
+    this.sortedPropertiesByRequired = {
+      ...requiredProperties,
+      ...optionalProperties,
+    };
+  }
+
+  private buildForm() {
+    this.createFormControlsWithTheirValidators();
+    this.form.markAllAsTouched();
+  }
+
+  private createFormControlsWithTheirValidators() {
+    createFormControlsWithTheirValidators(this.fb,this.propertiesMap,this.form,this.input,this.customizedInputs)
```

Comment:
This function is one line why not delete it and inline it?

Reply:
yeah will do that.